nimx/resource_cache.nim

```diff
@@ -15,6 +15,7 @@ type ResourceLoader* = ref object
     loadedSize: int
     itemsToLoad: int
     onComplete*: proc()
+    onProgress*: proc(name: string)
```

Comment:
There should be some smarter onProgress handler. Specifically, it should have a float progress :)